src/hooks/transactions/useEstimateFee.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { useCallback } from 'react';
+import { useApiCallsContext } from '../../contexts/ApiCallsContextProvider';
+import { useSourceTarget } from '../../contexts/SourceTargetContextProvider';
+import { TransactionActionCreators } from '../../actions/transactionActions';
+import { useUpdateTransactionContext, useTransactionContext } from '../../contexts/TransactionContext';
+import { getSubstrateDynamicNames } from '../../util/getSubstrateDynamicNames';
+import useLaneId from '../chain/useLaneId';
+
+export const useEstimateFee = () => {
```

Comment:
I would alter the useEstimateFee to follow the pattern mentioned [here](https://gist.github.com/tomusdrw/6be7adb067ba44920931d8146f7467f7#gistcomment-3769751) and a real-code example [here](https://github.com/paritytech/parity-bridges-ui/pull/190/files#diff-7ad0d602f18b557f8ef933910d583aee088359750a74fa8e12e69e0095177e12).

To explain: This hook returns a function - It can be instead converted in 2 different functions. The `getEstimateFeeCall` and a hook that will include the `getEstimatedFeeCall` (of course you will need to bring back the useGetApi hook that was removed) (see examples above). 

This is a proposal for breaking this hook to smaller layers - removing the dispatchers from the function logic (`layers 2-3`) - but again this (as is) can be assumed as `Layer 4` (see [here](https://gist.github.com/tomusdrw/6be7adb067ba44920931d8146f7467f7#gistcomment-3777140) for layers)

Reply:
hey ... please check the code now. Is not exactly the pattern you have shared , but is much similar.